src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/__init__.py

```diff
@@ -4,19 +4,24 @@
 # --------------------------------------------------------------------------------------------
 
 from azure.cli.core import AzCommandsLoader
-from azure.cli.core.profiles import ResourceType
+from azure.cli.core.profiles import ResourceType, CustomResourceType
 from azure.cli.core.commands import AzCommandGroup, AzArgumentContext
 
 import azure.cli.command_modules.storage._help  # pylint: disable=unused-import
 
 
+CUSTOM_MGMT_STORAGE = CustomResourceType('azure.mgmt.storage', 'StorageManagementClient')
+
+
 class StorageCommandsLoader(AzCommandsLoader):
     def __init__(self, cli_ctx=None):
         from azure.cli.core.commands import CliCommandType
 
         storage_custom = CliCommandType(operations_tmpl='azure.cli.command_modules.storage.custom#{}')
+        custom_resource_types = {'latest': {CUSTOM_MGMT_STORAGE: '2015-06-15'}}
```

Comment:
This structure isn't really intuitive.  It should be validated in the __init__. Otherwise an error will likely result in a stack trace.

Reply:
This will be resolved with Willie's suggestion.